inc/admin.php

```diff
@@ -751,8 +751,8 @@ function update_home_on_theme_change() {
 	function get_widgets() {
 		global $wp_widget_factory;
 		$widgets = array();
-		foreach ( $wp_widget_factory->widgets as $widget_obj ) {
-			$class = get_class( $widget_obj );
+		foreach ( $wp_widget_factory->widgets as $class => $widget_obj ) {
+			$class = preg_match( '/[0-9a-f]{32}/', $class ) ? get_class( $widget_obj ) : $class;
```

Comment:
The only question is whether the hashes are guaranteed, currently and in future, to be 32 character strings. E.g. what happens if they decide to start using `spl_object_id` (available in PHP 7.2) instead of `spl_object_hash` to generate the hashes? And of course, the fairly unlikely, but still valid case of having a class name containing only the above characters and having a length of 32. Love PHP. ❤️ 🤷‍♂

Reply:
From a little bit of testing and a few examples online, it seems like the hash is always 32 character hex.

If WordPress or PHP change anything in the future, then we can just expand the regex to pick that up. For now, it's a lot more robust than trying to identify something that looks like a class name.